modules/widgets/recurring-payments.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * Disable direct access/execution to/of the widget code.
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'Jetpack_Recurring_Payments_Widget' ) ) {
+	/**
+	 * Recurring Payments Button
+	 *
+	 * Display a Recurring Payments Button as a Widget.
+	 */
+	class Jetpack_Recurring_Payments_Widget extends WP_Widget {
+
+		/**
+		 * Constructor.
+		 */
+		function __construct() {
+			parent::__construct(
+				'jetpack_recurring_payments_widget',
+				/** This filter is documented in modules/widgets/facebook-likebox.php */
+				apply_filters( 'jetpack_widget_name', __( 'Recurring Payments', 'jetpack' ) ),
+				array(
+					'classname'                   => 'jetpack-recurring-payments',
+					'description'                 => __( 'Add a Recurring Payments Button as a Widget.', 'jetpack' ),
+					'customize_selective_refresh' => true,
+				)
+			);
+
+		}
+
+		/**
+		 * Return an associative array of default values.
+		 *
+		 * These values are used in new widgets.
+		 *
+		 * @return array Default values for the widget options.
+		 */
+		private function defaults() {
+			$first_product = $this->get_first_product();
+
+			return array(
+				'title'                    => '',
+				'product_post_id'          => $first_product['id'],
```

Comment:
What happens here if `$this->get_first_product()` returned `null`?

Reply:
Nice catch!
I refactored it a bit and now it takes this into account much better